pgjdbc/src/main/java/org/postgresql/jdbc/PgDatabaseMetaData.java

```diff
@@ -2671,7 +2671,9 @@ public ResultSet getUDTs(@Nullable String catalog, @Nullable String schemaPatter
     for (Iterator<Integer> i = connection.getTypeInfo().getPGTypeOidsWithSQLTypes(); i.hasNext(); ) {
       Integer typOid = i.next();
       int sqlType = connection.getTypeInfo().getSQLType(typOid);
-      sqlwhen.append(" when t.oid = ").append(typOid).append(" then ").append(sqlType);
+      // Note: Java Integers are signed 32-bit integers, but oids are unsigned 32-bit integers.
+      // We must therefore map it to a positive long value before writing it into the query, or we'll be unable to correctly handle ~ half of the oid space.
+      sqlwhen.append(" when t.oid = ").append(connection.getTypeInfo().intOidToLong(typOid)).append(" then ").append(sqlType);
```

Comment:
Probably is more readable to move up the  intOidToLong() method call, and have something like
`long typOid = connection.getTypeInfo().intOidToLong(i.next());` instead of having it on the append.

Reply:
+1 to breaking the code up to be more readable

Reply:
I'll push an update with this and the getSQLType()-test shortly.

Reply:
Update pushed.